authentication/auth.go

```diff
@@ -556,17 +556,13 @@ func GenerateTokenFromDefaultsV2(schemaURL string, accountServiceURL string, url
 	return token, ""
 }
 
-func addAdditionalMetadata(surveyMetadata map[string]interface{}, requiredSchemaMetadata []Metadata) []Metadata {
-	for _, metadataVal := range surveyMetadata {
-		for key, value := range metadataVal.(map[string]interface{}) {
-			if isExistingRequiredSchemaMetadata(requiredSchemaMetadata, key) {
-				continue
-			} else {
-				newMetadata := Metadata{}
-				if strings.Contains(value.(string), "True") || strings.Contains(value.(string), "False") {
-					newMetadata = Metadata{Name: key, Validator: "boolean", Default: "false"}
-					requiredSchemaMetadata = append(requiredSchemaMetadata, newMetadata)
-				}
+func addUrlBooleanMetadata(updatedMetadata map[string]interface{}, requiredSchemaMetadata []Metadata) []Metadata {
+	for key, value := range updatedMetadata {
+		if !(isExistingRequiredSchemaMetadata(requiredSchemaMetadata, key)) {
+			value = strings.ToLower(value.(string))
+			if strings.Contains(value.(string), "true") || strings.Contains(value.(string), "false") {
```

Comment:
Not only it contains but I would say it equals, so use the equals method in Go, something like:
```go
var testedString = strings.ToLower(value.(string))
strings.EqualFold(testedString, "true")
```

Reply:
Nice spot! Contains could have lead to some issues further on, updated 👍